router/proxy/transport.go

```diff
@@ -50,6 +54,109 @@ type transport struct {
 
 	stickyCookieKey   *[32]byte
 	useStickySessions bool
+
+	inFlightMtx      sync.Mutex
+	inFlightRequests map[string]int64
+}
+
+func (t *transport) trackRequestStart(backend *router.Backend) {
+	t.inFlightMtx.Lock()
+	defer t.inFlightMtx.Unlock()
```

Comment:
`defer` has some overhead and this is a hot path, just unlock it after incrementing (make the same change in `trackRequestEnd`.

Reply:
Updated in d71e611.